src/main/java/org/esbtools/gateway/resync/controller/ResyncGateway.java

```diff
@@ -0,0 +1,74 @@
+package org.esbtools.gateway.resync.controller;
+
+import org.esbtools.gateway.resync.ResyncRequest;
+import org.esbtools.gateway.resync.ResyncResponse;
+import org.esbtools.gateway.resync.exception.IncompleteRequestException;
+import org.esbtools.gateway.resync.exception.InvalidSystemException;
+import org.esbtools.gateway.resync.exception.ResyncFailedException;
+import org.esbtools.gateway.resync.exception.SystemConfigurationException;
+import org.esbtools.gateway.resync.service.ResyncService;
+import org.esbtools.gateway.resync.service.ResyncServiceRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/")
+public class ResyncGateway {
+
+    protected ResyncServiceRepository resyncServiceRepository;
+
+    @Autowired
+    public ResyncGateway(ResyncServiceRepository resyncServiceRepository) {
+        this.resyncServiceRepository = resyncServiceRepository;
+    }
+
+    @RequestMapping(value="/resync", method=RequestMethod.POST, produces="application/json")
+    public ResponseEntity<ResyncResponse> resync(@RequestBody ResyncRequest resyncRequest) {
+        ResponseEntity<ResyncResponse> responseEntity;
+        try {
+            ResyncService resyncService = resyncServiceRepository.getBySystem(resyncRequest.getSystem());
+            ResyncResponse resyncResponse = resyncService.resync(resyncRequest);
+            responseEntity = new ResponseEntity<ResyncResponse>(resyncResponse, HttpStatus.OK);
+        } catch(InvalidSystemException e) {
+            responseEntity = invalidSystemExceptionHandler(e);
+        } catch (SystemConfigurationException e) {
+            responseEntity = systemConfigurationExceptionHandler(e);
+        } catch(IncompleteRequestException e) {
+            responseEntity = incompleteRequestExceptionHandler(e);
+        } catch(ResyncFailedException e) {
+            responseEntity = resyncFailedExceptionHandler(e);
+        }
```

Comment:
So I *think* you should be able to remove these catch blocks and the `@ExceptionHandler` annotated methods should kick in appropriately. At least I hope so, because that would be pretty rad :sunglasses: 

That will change testing again however, and you might end up needing to test that using the MVC test framework to make sure it works. I would be fine with merging this PR before that, if you wanted to work on doing that separately. Up to you. Also I can help out if you want.